skfda/clustering.py

```diff
@@ -0,0 +1,1180 @@
+"""Clustering Module."""
+
+import numpy as np
+from .grid import FDataGrid
+from .metrics import pairwise_distance, lp_distance, norm_lp
+import matplotlib.pyplot as plt
+from mpldatacursor import datacursor
+import matplotlib.patches as mpatches
+from matplotlib.ticker import MaxNLocator
+import random
+from abc import ABC, abstractmethod
+
+__author__ = "Amanda Hernando Bernabé"
+__email__ = "amanda.hernando@estudiante.uam.es"
+
+
+class ClusteringData(ABC):
+    """Base class to implement clustering algorithms.
+
+    Class from which both :class:`K-Means <fda.clustering.KMeans>` and
+    :class:`Fuzzy K-Means <fda.clustering.FuzzyKMeans>` classes inherit."""
+
+    def __init__(self, max_iter, random_state, p):
```

Comment:
Please, check all estimators with the utilities explained in https://scikit-learn.org/stable/developers/contributing.html#rolling-your-own-estimator, because I am not sure that you are following all the scikit-learn conventions.

Reply:
I have used `check_estimator` to see if the implemented estimators `KMeans` and `FuzzyKMeans` pass the test. It fails in both cases because the first parameter of the `fit` function expects a numpy.ndarray instead of an FDataGrid object., and in my opinion it should be like this.